students/TacSPx/lesson03/mailroom_modified.py

```diff
@@ -0,0 +1,166 @@
+#
+# <05/30/2020>, Created Script, mailroom part 1
+# <06/12/2020>, Clean-up to snake case, adding sorting in reporting descending totals, fix indentation error
+# ---------------------------------------------------------------------------- #
+#imports
+import sys
+from collections import defaultdict
+from operator import itemgetter
+# Data ---------------------------------------------------------------------- #
+# Task 1: Create data
+# Create a data structure that holds a list of your donors and a history of the amounts they have donated.
+# Have least five donors, with between 1 and 3 donations each, store in the global namespace.
+
+
+donor_info = [("Jim Zorn", [3772.32, 1201.17]),
+            ("Jermaine Kearse", [877.33]),
+            ("Marcus Trufant", [1563.23, 1043.87, 1.32]),
+            ("K.J. Wright", [21663.23, 300.87, 100432.0]),
+            ("Curt Warner", [663.23, 300.87, 10432.0]),
+            ]
+
+# Processing  --------------------------------------------------------------- #
+# Task 2:  Menu choices
+# Prompt the user to choose from a menu of 3 actions: “Send a Thank You”, “Create a Report” or “quit”.
+###############################################
+#   "Menu of Choices"
+###############################################
+
+def main():
+    '''
+    Menu of Choices:  Thank You, Report, Create a Report
+    :return:
+    '''
+
+    while (True):
+        print("""
+        \n**** XYZ Nonprofit Charity ****\n
+        Menu of Options
+        1) Send a "Thank You"
+        2) Create a Report
+        3) Exit Program
+        """)
+        str_choice = str(input("Which option would you like to perform? [1 to 3] - "))
+        print()  # added a new line for spacing looks
+        if (str_choice.strip() == '1'):
+            thank_you()
+        elif (str_choice.strip() == '2'):
+            report()
+        elif (str_choice.strip() == '3'):
+            exit_program()
+        else:
+            print("\nPlease choose only 1, 2, or 3!")
+
+###############################################
+#   #1 Send a "Thank You"
+###############################################
+def thank_you():
+    '''
+    Thank You:  Will append donations to existing donors, or adds new donors with donations.
+    User can see list of existing donors by typing 'list'.  Once donation is entered script will create
+    custom thank you letter/ email.
+    :return:
+    '''
+    while (True):
+        donor_name = input("Type 'List' to see all names in Database or\nType First and Last Name: ").title()
+        if (donor_name.lower() == 'list'):
+            # define format row
+            row = "{Name:<18s}".format
+            print("\nDONOR REGISTRY:")
+            donor_sort = (sorted(donor_info))
+            for donor in donor_sort:
+                donor_lst = (row(Name=donor[0]))
+                print(donor_lst)
+            input("Press Enter to Continue")
+            print()
+            continue
+
+        if (donor_name.lower() != 'list'):
```

Comment:
this should be an else statement for if statement on line 70 -- you should never have to test the same variable twice, if there are only two choices (either it's a "list" or not)